drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -446,6 +449,16 @@ local function create_thermostat_modes_profile(device)
 end
 
 local function match_profile(driver, device, battery_supported)
+  -- read WindSupport before profiling if the FanControl Wind Feature Flag is supported
+  local wind_eps = device:get_endpoints(clusters.FanControl.ID, { feature_bitmap = clusters.FanControl.types.FanControlFeature.WIND })
+  if #wind_eps > 0 and device:get_field(WIND_MODE_COUNT) == nil then
+    device:set_field(BATTERY_SUPPORT, battery_supported, {persist = true}) -- save value for after WindSupport read
+    local req = im.InteractionRequest(im.InteractionRequest.RequestType.READ, {})
```

Comment:
This could be simplified
```suggestion
    device:send(clusters.FanControl.attributes.WindSupport:read())
```